src/auro-carousel.js

```diff
@@ -73,6 +77,41 @@ class AuroCarousel extends LitElement {
     this.setScrollFlags(false);
     this.setUpIntersectionObserver();
     this.setUpResizeObserver();
+    if (this.hasAttribute('centerSelectedOnRender')) {
+      this.actionOnChildrenReady();
+    }
+  }
+
+  /**
+   * function handler for anything that happens when all its children is ready
+   * @return {void}
+   */
+  actionOnChildrenReady() {
+    const promises = [];
+
+    [...this.children].forEach((child) => {
+      // here we check the 'updateComplete' property of its child, so it is only works on lit-element.
+      // Also FYI this would only work if this component is imported AFTER of its child component.
+      // Otherwise child.updateComplete would be undefined even though it is a lit-element.
```

Comment:
```suggestion
      // Here is the check the 'updateComplete' property of its child. Only works on lit-element.
      // This only works if this component is imported AFTER its child component.
      // Otherwise child.updateComplete would be undefined.
```